docs/_templates/sidebarlogo.html

```diff
@@ -20,3 +20,5 @@
   height: 100%;
 }
 </style>
+
+<progress id="progressbar" value="0" max="100"></progress>
```

Comment:
brief thought to follow up later: Maybe its possible to add this progress bar page-wise into the header, and set ``value`` and ``max`` based on a counter in each chapter. A chapter with 5 sections would add one fifth to the progress bar per page read (while pace of progress bar for that fifth is still dependent on section length).